hack/jenkins/README.md

```diff
@@ -0,0 +1,28 @@
+# Jenkins
+
+[Jenkins](http://jenkins-ci.org/) is a pluggable continuous
+integration system. The Google team is running a Jenkins server on a
+private GCE instance for the Kubernetes project in order to run longer
+integration tests, continuously, on different providers. Currently, we
+(Google) are only running Jenkins on our own providers (GCE and GKE)
+in different flavors.
+
+## General flow
+The flow of the Google Jenkins server:
+* Under the `kubernetes-build` job: Every 5 minutes, Jenkins polls for a batch of new commits, after which it runs the `build.sh` script (in this directory) on the latest tip. This results in build assets getting pushed to GCS and the `latest.txt` file in the `ci` bucket being updated. That job then triggers `kubernetes-e2e-*`.
+* On trigger, and every half hour (which effectively means all the time, unless we're failing cluster creation), e2e variants run, on the latest build assets in GCS:
+  * `kubernetes-e2e-gce`: Standard GCE e2e
+  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s and GKE creating clusters at its default version
+  * `kubernetes-e2e-gke-ci`: GKE provider e2e, with head k8s and GKE creating clusters at the head k8s version
+* Each named job may run in parallel with the others, i.e. `kubernetes-build` with `kubernetes-e2e-gce`, but `kubernetes-e2e-gce` will not run concurrently with itself.
```

Comment:
Fair. I can put it back - it's interesting enough.

On Fri Jan 09 2015 at 4:15:48 PM davidopp <notifications@github.com> wrote:

> In hack/jenkins/README.md
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3360#discussion-diff-22752584>
> :
>
> > +
> > +[Jenkins](http://jenkins-ci.org/
>
> ) is a pluggable continuous
> > +integration system. The Google team is running a Jenkins server on a
> > +private GCE instance for the Kubernetes project in order to run longer
> > +integration tests, continuously, on different providers. Currently, we
> > +(Google) are only running Jenkins on our own providers (GCE and GKE)
> > +in different flavors.
> > +
> > +## General flow
> > +The flow of the Google Jenkins server:
> > +* Under the `kubernetes-build` job: Every 5 minutes, Jenkins polls for a batch of new commits, after which it runs the `build.sh` script (in this directory) on the latest tip. This results in build assets getting pushed to GCS and the `latest.txt` file in the `ci` bucket being updated. That job then triggers `kubernetes-e2e-*`.
> > +* On trigger, and every half hour (which effectively means all the time, unless we're failing cluster creation), e2e variants run, on the latest build assets in GCS:
> > +  * `kubernetes-e2e-gce`: Standard GCE e2e
> > +  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s and GKE creating clusters at its default version
> > +  * `kubernetes-e2e-gke-ci`: GKE provider e2e, with head k8s and GKE creating clusters at the head k8s version
> > +* Each named job may run in parallel with the others, i.e. `kubernetes-build` with `kubernetes-e2e-gce`, but `kubernetes-e2e-gce` will not run concurrently with itself.
>
> Well, it's up to you, but I suspect you only need a minor tweak to clearly
> described how Jenkins runs the stuff (you're 99% of the way there already).
> I think it would be of interest to people to know whether a burst of
> submissions means that the last one will not see Jenkins results for a long
> time or will see them right away (assuming sufficient resources).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3360/files#r22752584>
> .
>